block_enrol_student.php

```diff
@@ -0,0 +1,86 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/enrol/locallib.php');
+
+use block_enrol_student\output\email_list;
+
+/**
+ * Form for editing HTML block instances.
+ *
+ * @package    block_enrol_student
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+class block_enrol_student extends block_base {
+
+    public function init() {
+        $this->title = get_string('pluginname', 'block_enrol_student');
+    }
+
+    public function has_config() {
+        return true;
+    }
+
+    public function applicable_formats() {
+        return array('all' => true);
+    }
+
+    public function instance_allow_multiple() {
+        return true;
+    }
+
+    public function get_content() {
+        if (optional_param('course', 0, PARAM_INT) > 0) {
```

Comment:
It would be better assign the return value of optional_param('course', 0, PARAM_INT); to a variable and use it (if that function is called in many places with same params)